### PR TITLE
Add custom components for markdown hyperlinks and images

### PR DESCRIPTION
**Pull request by mattbowen-usds**
This adds two custom components and overrides hyperlinks and image tags in mdx files

### CODE REVIEW

File: tech-far-hub/package.json [hunk 1]
```diff
@@ -13,11 +13,16 @@
     "build": "gatsby build",
     "serve": "gatsby serve",
     "clean": "gatsby clean",
-    "typecheck": "tsc --noEmit"
+    "typecheck": "tsc --noEmit",
```
Comment (vim-usds) [inline tech-far-hub/package.json:16]: Nice! I just added this myself to the new repo.. although stuff is failing.. which means I might have to patch it

File: tech-far-hub/package.json [hunk 2]
```diff
@@ -13,11 +13,16 @@
...
+    "test:watch": "vitest watch",
```
Comment (vim-usds) [inline tech-far-hub/package.json:17]: I like watch mode.. it just keeps running!

File: tech-far-hub/package.json [hunk 3]
```diff
@@ -13,11 +13,16 @@
...
+    "test:ci": "vitest run",
+    "test:coverage": "vitest run --coverage"
```
Comment (vim-usds) [inline tech-far-hub/package.json:19]: I might need to add this!

File: tech-far-hub/package.json [hunk 4]
```diff
@@ -33,18 +38,25 @@
...
     "ts-deep-pick": "^0.2.2",
     "uswds": "^2.13.3"
   },
   "devDependencies": {
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
     "@types/node": "^18.11.9",
     "@types/react": "^18.0.25",
     "@types/react-dom": "^18.0.9",
+    "c8": "^7.12.0",
```
Comment (vim-usds) [inline tech-far-hub/package.json:54]: Had to look this up! Nice find.. would love to know how actionable the coverage reports are

File: tech-far-hub/src/components/__tests__/hyperlink.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    siteMetadata: {
+      siteTitle: `Testing Hyperlinks`,
+      siteUrl: `https://example.com`,
+    },
+  },
+};
+
+describe("Hyperlink comonent", () => {
+  beforeEach(() => {
+    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
```
Comment (vim-usds) [inline tech-far-hub/src/components/__tests__/hyperlink.tsx:19]: Nice mocking here!

File: tech-far-hub/src/pages/index.tsx [hunk 6]
```diff
@@ -55,10 +55,10 @@ const IndexPage: React.FC<PageProps<Queries.HomePageInitiativesQuery>> = ({
           <h3>01. Get Started</h3>
           <ol>
             <li>
-              <a href="/get-started/">What is the TechFAR Hub?</a>
+              <a href="get-started/">What is the TechFAR Hub?</a>
```
Comment (vim-usds) [inline tech-far-hub/src/pages/index.tsx:58]: what allows removing the prefix slash?

File: tech-far-hub/src/url-utils.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import { IURLInfo, DownloadFileType } from "./types";
+
+export class URLInfo implements IURLInfo {
```
Comment (vim-usds) [inline tech-far-hub/src/url-utils.tsx:3]: class components!

File: tech-far-hub/vitest-setup.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { vi } from "vitest";
+import React from "react";
+import "@testing-library/jest-dom";
+
+vi.mock(`gatsby`, async () => {
+  const gatsby = await vi.importActual<typeof import("gatsby")>(`gatsby`);
```
Comment (vim-usds) [inline tech-far-hub/vitest-setup.ts:6]: I might have to take this!

**Review by vim-usds — APPROVED** [review]
So this PR didn't work?

**Comment by mattbowen** [comment]
@vim-usds the PR did --- I just had an even worse idea that I'm not gonna pursue

File: tech-far-hub/src/pages/index.tsx (hunk 6 above)
Comment (mattbowen) [reply tech-far-hub/src/pages/index.tsx:58]: it's the homepage, so I don't need it

File: tech-far-hub/src/url-utils.tsx (hunk 7 above)
Comment (mattbowen) [reply tech-far-hub/src/url-utils.tsx:3]: I was so happy to just write like a normal class tbh

**Pull request merged**